packages/samples/headless-react/src/components/numeric-facet/numeric-facet.class.tsx

```diff
@@ -0,0 +1,91 @@
+import {Component} from 'react';
+import {
+  buildNumericFacet,
+  NumericFacet as HeadlessNumericFacet,
+  NumericFacetState,
+  NumericFacetValue,
+  NumericRangeRequest,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface BaseNumericFacetProps {
+  field: string;
+  facetId: string;
+  format: (n: number) => string;
+}
+
+interface AutomaticNumericFacetProps extends BaseNumericFacetProps {
+  generateAutomaticRanges: true;
+  currentValues?: NumericRangeRequest[];
+}
+
+interface ManualNumericFacetProps extends BaseNumericFacetProps {
+  generateAutomaticRanges: false;
+  currentValues: NumericRangeRequest[];
+}
```

Comment:
I'd remove these interfaces in favor of the `NumericFacetProps` exported by headless. Devs should not need to create `Automatic` and `Manual` interfaces in their code.

Reply:
The common pattern with the samples so far was to redefine the interfaces since `*Props` interfaces typically contain `initialState`, `options` and nothing else, so for now I'll just remove the discriminating union.

However, we could create a story after the samples are done to change them all to use the `*Props` or `*Options` interfaces.

Reply:
That's true, I should have written the `NumericFacetOptions` interface. Could we use that instead?